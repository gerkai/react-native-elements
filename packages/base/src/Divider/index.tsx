import { Divider, DividerProps } from './Divider';
import { withTheme } from '../config';

export { Divider };
export type { DividerProps };
export default withTheme(Divider, 'Divider');
