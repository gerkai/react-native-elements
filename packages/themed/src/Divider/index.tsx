import {
  Divider,
  DividerProps,
} from '@react-native-elements/base/dist/Divider/Divider';
import { withTheme } from '../config';

export { Divider };
export type { DividerProps };
export default withTheme(Divider, 'Divider');
