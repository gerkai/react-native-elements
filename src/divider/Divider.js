import React from 'react';
import { View, StyleSheet } from 'react-native';
import ViewPropTypes from '../config/ViewPropTypes';

const Divider = ({ style }) => <View style={[styles.container, style]} />;

const hairlineWidth = StyleSheet.hairlineWidth;

Divider.propTypes = {
  style: ViewPropTypes.style,
};

const styles = StyleSheet.create({
  container: {
    // Darker color if hairlineWidth is not thin enough
    backgroundColor: hairlineWidth < 1 ? '#BCBBC1' : 'rgba(0, 0, 0, 0.12)',
    height: hairlineWidth,
  },
});

export default Divider;
