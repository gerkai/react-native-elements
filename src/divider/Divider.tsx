import React from 'react';
import { View, StyleSheet, ViewProps } from 'react-native';
import { withTheme } from '../config';
import { RneFunctionComponent } from '../helpers';

export type DividerProps = ViewProps & {
  style?: object | any[];
};

const Divider: RneFunctionComponent<DividerProps> = ({
  style,
  theme,
  ...rest
}) => (
  <View
    style={StyleSheet.flatten([
      {
        backgroundColor: theme?.colors?.divider,
        height: StyleSheet.hairlineWidth,
      },
      style,
    ])}
    {...rest}
  />
);

export { Divider };
export default withTheme(Divider, 'Divider');
