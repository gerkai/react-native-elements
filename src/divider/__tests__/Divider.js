import React from 'react';
import {shallow} from 'enzyme';
import Divider from '../Divider';

describe('Divider Component', () => {
  it('should render without issues', () => {
    const component = shallow(<Divider />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });
  it('should render with style', () => {
    const component = shallow(<Divider style={{ backgroundColor: 'blue' }} />);

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
    expect(component.props().style.length).toBe(2);
    expect(component.props().style[1].backgroundColor).toBe('blue');
  });
});
