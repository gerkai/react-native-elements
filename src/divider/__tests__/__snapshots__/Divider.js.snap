// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component should render with style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      },
      Object {
        "backgroundColor": "blue",
      },
    ]
  }
/>
`;

exports[`Divider Component should render without issues 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#BCBBC1",
        "height": 0.5,
      },
      undefined,
    ]
  }
/>
`;
